plugins/woocommerce/includes/wc-cart-functions.php

```diff
@@ -94,7 +94,7 @@ function wc_get_raw_referer() {
  *
  * @return mixed
  */
-function wc_add_to_cart_message( $products, $show_qty = false, $return = false ) {
+function wc_add_to_cart_message( $products, $show_qty = false, $return = false, $variation_id = false ) {
```

Comment:
If the above suggestion is applied, then no need to send in `$variation_id` as a dedicated param. Although we would want to refactor `$products` param like so:
```suggestion
function wc_add_to_cart_message( $products_or_variations, $show_qty = false, $return = false ) {
```

Note that I have not added it in the suggestion, but we would want to refactor variable names everywhere in the function.

Reply:
In terms of functionality the initial change works, and generates the expected result. The issue is mainly about the display of the title in the message, so it'd be easier to just pass the correct ID to the `get_the_title` function, and leave the filters as they are. Refactoring the filters is going to be complicated if they're expecting the Parent ID, rather than Variation ID. Maybe multiple variations should be supported as well, but in terms of fixing the bug, displaying the correct title would work in my opinion. 

```suggestion
function wc_add_to_cart_message( $products, $show_qty = false, $return = false, $variation_id = false ) {

// Existing code here
// Line 113
    !empty( $variation_id ) ? $title_id = $variation_id : $title_id = $product_id;

    $titles[] = apply_filters( 'woocommerce_add_to_cart_qty_html', ( $qty > 1 ? absint( $qty ) . ' &times; ' : '' ), $product_id ) . apply_filters( 'woocommerce_add_to_cart_item_name_in_quotes', sprintf( _x( '&ldquo;%s&rdquo;', 'Item name in quotes', 'woocommerce' ), strip_tags( get_the_title( $title_id ) ) ), $product_id );
```